### PR TITLE
IDM unit tests

### PR DESCRIPTION
**Pull request by maqi**
This change is [Reviewable](https://reviewable.io/reviews/maidsafe/safe_vault/376)

### CODE REVIEW

**Comment by maidsafe-highfive** [comment]
r? @Fraser999

(maidsafe_highfive has picked a reviewer for you, use r? to override)

File: src/personas/immutable_data_manager.rs [hunk 1]
```diff
@@ -853,19 +870,240 @@ mod test {
...
+            panic!("Received unexpected response {:?}", get_success[0]);
+        }
+    }
+
+    #[test]
+    fn get_after_put_success() {
+        let mut env = Environment::new();
+        let put_env = env.put_im_data();
+        let put_requests = env.routing.put_requests_given();
+        let data_hodlers : Vec<XorName> = put_requests.iter().map(|put_request| {
```
Comment (Fraser999) [inline src/personas/immutable_data_manager.rs]: Typo in `data_hodlers` - this is copied across a few tests.

File: src/personas/immutable_data_manager.rs [hunk 2]
```diff
@@ -853,19 +870,240 @@ mod test {
...
+    #[test]
+    fn handle_put_failure() {
+        let mut env = Environment::new();
+        let put_env = env.put_im_data();
+        let put_requests = env.routing.put_requests_given();
+        let data_hodlers : Vec<XorName> = put_requests.iter().map(|put_request| {
+            put_request.dst.name().clone()
+        }).collect();
+
+        let mut candidates = data_hodlers.clone();
```
Comment (Fraser999) [inline src/personas/immutable_data_manager.rs]: This variable name is a bit misleading - they're not candidates, more like `current_holders` or something?

File: src/personas/immutable_data_manager.rs [hunk 3]
```diff
@@ -853,19 +870,240 @@ mod test {
...
+        // The current setting is REPLICANTS = MIN_REPLICANTS = 4
+        // which means any failure notification from PM will trigger a replication
```
Comment (Fraser999) [inline src/personas/immutable_data_manager.rs]: As your comment hints, this test will start failing if `MIN_REPLICANTS` is changed to be less than `REPLICANTS`.  Can this be fixed please?

**Comment by Fraser999** [comment]
Reviewed 1 of 1 files at r2.
Review status: all files reviewed at latest revision, 3 unresolved discussions.

---



*Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/maidsafe/safe_vault/376)*

**Comment by Fraser999** [comment]
Reviewed 1 of 1 files at r3.
Review status: all files reviewed at latest revision, all discussions resolved.

---



*Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/maidsafe/safe_vault/376)*

**Pull request merged**